lib/folio_record.rb

```diff
@@ -260,12 +262,33 @@ def stripped_marc_json
     end
   end
 
-  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
   def instance_derived_marc_record
     MARC::Record.new.tap do |marc|
       marc.append(MARC::ControlField.new('001', hrid))
       # mode of issuance
       # identifiers
+      record.dig('instance', 'identifiers').each do |identifier|
+        id_value = identifier['value']
+
+        case id_value
+        # LCCN
+        when /^([ a-z]{0,3}\d{8} |[ a-z]{0,2}\d{10})$/
```

Comment:
I just noted on the ticket, we have real-world 010 values like
   68038902 //r84

that won't be caught by this revised regex.

Reply:
It would be great if we could extract this to it's own class (e.g. `IdentifierClassifier` maybe?).  That would make it much easier to write tests for.

Reply:
Should the extra data at the end of LCCNs be preserved as the identifier value? I.e. keep the entire ` 68038902 //r84` and not just ` 68038902`

Reply:
Perfect, that keeps it simple

Reply:
Updated to remove the terminator and require (vs. just allow) the blank/alpha prefixes to avoid confusing with ISBNs.

Reply:
@jcoyne where would you want that to live?

Reply:
I suggest we handle extracting classes in a separate PR. Who knows, we might discover we need to pull the identifier type info anyway 🤷‍♂️